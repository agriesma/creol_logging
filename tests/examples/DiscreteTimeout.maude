load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "DiscreteTimeout" : Class | Inh: noInh, Param: "other", Att: "other"
      |-> null, "clock" |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: assign( ( "clock" @ "DiscreteTimeout" ) ; int(0) ) ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "ccglab0" |-> null, "_" |->
              null, Code: assign( "clock" ; "+" ( "clock" :: int(1) ) ) ;
              call( "ccglab0" ; "this" ; "run" ; emp ) ; free( "ccglab0" ) ;
              return ( emp ) >,
            < "timeout" : Method | Param: "n", Att: "n" |-> null, "result"
              |-> null, "timeout" |-> null, "l" |-> null, "_" |-> null,
              Code: assign( "timeout" ; "+" ( "clock" :: int(10) ) ) ;
              call( "l" ; "other" ; "m" ; emp ) ;
              (await "&&" ( ?("l") :: "<=" ( "clock" :: "timeout" ) ) ;
                get( "l" ; noVid ) ; free( "l" ) ;
                assign( "result" ; bool(true) ) [] await ">" ( "clock" ::
                "timeout" ) ; free( "l" ) ; assign( "result" ; bool(false) ))
              ; return ( "result" ) >,
      Ocnt: 0 > } .
endm

