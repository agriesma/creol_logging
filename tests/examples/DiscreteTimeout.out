==========================================
reduce in PROGRAM : state .
result State: {
< "DiscreteTimeout" : Class | Inh: noInh, Param: "other", Att: "clock" |->
    null, "other" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assign("clock" @
    "DiscreteTimeout" ; int(0)) ; return(emp) >, < "run" : Method | Param:
    noVid, Att: "_" |-> null, "ccglab0" |-> null, Code: assign("clock" ; "+"(
    "clock" :: int(1))) ; call("ccglab0" ; "this" ; "run" ; emp) ; free(
    "ccglab0") ; return(emp) >, < "timeout" : Method | Param: "n", Att: "_" |->
    null, "l" |-> null, "n" |-> null, "result" |-> null, "timeout" |-> null,
    Code: assign("timeout" ; "+"("clock" :: int(10))) ; call("l" ; "other" ;
    "m" ; emp) ; (await ">"("clock" :: "timeout") ; free("l") ; assign("result"
    ; bool(false))[]await "&&"(?("l") :: "<="("clock" :: "timeout")) ; get("l"
    ; noVid) ; free("l") ; assign("result" ; bool(true))) ; return("result") >,
  Ocnt: 0 >
}
Bye.
